lib/spack/spack/cmd/extensions.py

```diff
@@ -76,37 +102,46 @@ def extensions(parser, args):
     if not args.mode:
         args.mode = 'short'
 
-    #
-    # List package names of extensions
-    extensions = spack.repo.extensions_for(spec)
-    if not extensions:
-        tty.msg("%s has no extensions." % spec.cshort_spec)
-        return
-    tty.msg(spec.cshort_spec)
-    tty.msg("%d extensions:" % len(extensions))
-    colify(ext.name for ext in extensions)
-
-    #
-    # List specs of installed extensions.
-    #
-    installed = [s.spec
-                 for s in spack.store.db.installed_extensions_for(spec)]
-
-    print
-    if not installed:
-        tty.msg("None installed.")
-        return
-    tty.msg("%d installed:" % len(installed))
-    spack.cmd.find.display_specs(installed, mode=args.mode)
-
-    #
-    # List specs of activated extensions.
-    #
-    activated = spack.store.layout.extension_map(spec)
-    print
-    if not activated:
-        tty.msg("None activated.")
-        return
-    tty.msg("%d currently activated:" % len(activated))
-    spack.cmd.find.display_specs(
-        activated.values(), mode=args.mode, long=args.long)
+    if show_packages:
+        #
+        # List package names of extensions
+        extensions = spack.repo.extensions_for(spec)
+        if not extensions:
+            tty.msg("%s has no extensions." % spec.cshort_spec)
+        else:
+            tty.msg(spec.cshort_spec)
+            tty.msg("%d extensions:" % len(extensions))
+            colify(ext.name for ext in extensions)
+
+    layout = spack.store.extensions
+    if args.view is not None:
+        layout = YamlViewExtensionsLayout(args.view, spack.store.layout)
+
+    if show_installed:
+        #
+        # List specs of installed extensions.
+        #
+        installed = [s.spec
+                     for s in spack.store.db.installed_extensions_for(spec)]
+
+        if show_all:
+            print
```

Comment:
what was the idea here @mathstuf?

Reply:
Uhh…probably either stray debugging (unlikely) or an incomplete/untested flag combination.